docs/core-concepts/dotnet-core.md

```diff
@@ -0,0 +1,145 @@
+
+The product is broken into several pieces, enabling the various parts to be adapted to new platforms on different schedules. The runtime and platform-specific foundational libraries must be ported as a unit. Platform-agnostic libraries should work as-is on all platforms, by construction. To a large degree, this split is what makes the .NET team efficient. The team invests in reducing platform-specific implementations, prefering platform-neutral C# code whenever an algorithm or API can be implemented in-full or in-part that way.
+
+People commonly ask how .NET Core is implemented, in order to support multiple operating systems. They typically ask if there are separate implementations or if [conditional compilation](https://en.wikipedia.org/wiki/Conditional_compilation) is used. It's both, with a strong bias towards conditional compilation.
+
+You can see in the chart below that the vast majority of [CoreFX](https://github.com/dotnet/corefx) is platform-neutral code that is shared across all platforms. Windows and unix implementations are similar in size. Windows has a larger implementation since CoreFX implements some Windows-only features, such as [Microsoft.Win32.Registry](https://github.com/dotnet/corefx/tree/master/src/Microsoft.Win32.Registry) and [System.Security.Principal.Windows](https://github.com/dotnet/corefx/tree/master/src/System.Security.Principal.Windows) but does not yet implement any unix-only concepts. You will also see that the majority of the Linux and OS X implementations are shared across a unix implementation, while the Linux- and OS X-specific implementations are roughly similar in size.
+
+![CoreFX: Lines of Code per Platform](../corefx-platforms-loc.png)
+
+There are a mix of platform-specific and platform-neutral libraries in .NET Core. You can see the pattern in a few examples:
+
+- [CoreCLR](https://github.com/dotnet/coreclr) is platform-specific. It's built in C/C++, so is platform-specific by construction.
+- [System.IO](https://github.com/dotnet/corefx/tree/master/src/System.IO)) and [System.Security.Cryptography.Algorithms](https://github.com/dotnet/corefx/tree/master/src/System.Security.Cryptography.Algorithms) are platform-specific, given that the storage and cryptography APIs differ significantly on each OS. 
+- [System.Collections](https://github.com/dotnet/corefx/tree/master/src/System.Collections)) and [System.Linq](https://github.com/dotnet/corefx/tree/master/src/System.Linq)) are platform-neutral, given that they create and operate over data structures.
```

Comment:
I'd prefer referencing System.Collections.Generic just to point readers to a more modern implementation.